internal/transport/http2_client.go

```diff
@@ -744,10 +748,10 @@ func (t *http2Client) NewStream(ctx context.Context, callHdr *CallHdr) (*Stream,
 	}
 	for {
 		success, err := t.controlBuf.executeAndPut(func(it interface{}) bool {
-			if !checkForStreamQuota(it) {
+			if !checkForHeaderListSize(it) {
 				return false
 			}
-			if !checkForHeaderListSize(it) {
+			if !checkForStreamQuota(it) {
 				return false
 			}
 			return true
```

Comment:
This can all be simply `return checkForHeaderListSize(it) && checkForStreamQuota(it)`

Reply:
Switched.